### PR TITLE
requestAnimationFrame addition to utils.

### PR DESCRIPTION
**Pull request by agamemnus**
Sorry -- idk how to add to a PR.

### CODE REVIEW

File: src/util.js [hunk 1]
```diff
@@ -1,5 +1,14 @@
 /* Common utilities */
 WaveSurfer.util = {
+    requestAnimationFrame: (
+        window.requestAnimationFrame ||
+        window.webkitRequestAnimationFrame ||
+        window.mozRequestAnimationFrame ||
+        window.oRequestAnimationFrame ||
+        window.msRequestAnimationFrame ||
+        function (callback, element) {window.setTimeout(callback, 1000 / 60);}
```
Comment (katspaugh) [inline src/util.js]: Thanks! Could you add spaces inside the anonymous function? And no need for `window`, `setTimeout` is a global also in Node.

File: src/util.js [hunk 2]
```diff
@@ -6,9 +6,18 @@ WaveSurfer.util = {
...
         window.oRequestAnimationFrame ||
         window.msRequestAnimationFrame ||
-        function (callback, element) {window.setTimeout(callback, 1000 / 60);}
+        function (callback, element) { setTimeout(callback, 1000 / 60); }
     ),
 
+    frame: function (func) {
+        return function () {
+            var my = this, args = arguments;
+            requestAnimationFrame(function () {
```
Comment (katspaugh) [inline src/util.js]: I guess, it should be `WaveSurfer.util.requestAnimationFrame` defined above?

**Comment by thijstriemstra** [comment]
ping @agamemnus

**Comment by agamemnus** [comment]
Oh sorry. Yes. Fixed.

File: src/util.js [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+        window.oRequestAnimationFrame ||
+        window.msRequestAnimationFrame ||
+        function (callback, element) { setTimeout(callback, 1000 / 60); }
+    ),
+
+    frame: function (func) {
+        return function () {
+            var my = this, args = arguments;
+            WaveSurfer.util.requestAnimationFrame(function () {
+                func.apply (my, args);
```
Comment (katspaugh) [inline src/util.js]: No need for the space after `apply`

Comment (agamemnus) [inline src/util.js]: No need for semicolons either...  :P

**Comment by katspaugh** [comment]
Merged, thanks much!

**Comment by agamemnus** [comment]
There is a problem that I just realized. Now this render is asynchronous, so anything using it actually needs a callback, etc.

**Comment by agamemnus** [comment]
@thijstriemstra 

You can't do drawWave anymore, or whatever. you have to add a callback parameter, because the function will run asynchronously -- on the next animation frame.

There are several dozen lines of code that need to be reworked. It is all interwoven... check out drawPeaks, for example...

https://github.com/agamemnus/wavesurfer.js/blob/master/dist/wavesurfer.js

**Comment by agamemnus** [comment]
(But then everything behind that needs callbacks too. So you have a callback pattern for drawPeaks (where I put the frame() call, and then callback pattern for drawBuffer. And then, although I didn't implement it, you need a callback for everything that calls drawBuffer.

**Comment by agamemnus** [comment]
@katspaugh SOS

**Comment by thijstriemstra** [comment]
A pr that reverts your changes might be necessary then.

**Comment by agamemnus** [comment]
Or just add the callback methods. But maybe for next because this will
break outside code that expects an instant change.

**Pull request merged**